### PR TITLE
Loki: Remove`TimeSrv` dependency

### PR DESCRIPTION
**Pull request by ivanahuckova**
Part of https://github.com/grafana/grafana/issues/72631 and the the **last step** 🎉 in removing `TimeSrv` dependency. In this PR I am:

- removing `TimeSrv` depenedency from Loki
- removing `getTimeRange` that was calling `timeSrv.geTimeRange()`
- `getTimeRangeParams` has now mandatory `timeRange: TimeRange` param
- `getDataSamples` has now mandatory `timeRange: TimeRange` param. In places where we are calling `getDataSamples`, we either pass provided `timeRange` (should be most of the times), or use `getDefaultTimeRange` from `@grafana/data`.

> [!TIP] 
> ### How to test
> 1. Run Loki `make devenv sources=loki`
> 2. Ensure that in query builder and query editor, the provided options when using `unwrap` are there. 
> image
> image

### CODE REVIEW

File: public/app/plugins/datasource/loki/datasource.ts [hunk 1]
```diff
@@ -150,8 +149,7 @@ export class LokiDatasource
 
   constructor(
     private instanceSettings: DataSourceInstanceSettings<LokiOptions>,
-    private readonly templateSrv: TemplateSrv = getTemplateSrv(),
-    private readonly timeSrv: TimeSrv = getTimeSrv()
```
Comment (matyax) [inline public/app/plugins/datasource/loki/datasource.ts:154]: Bye

File: public/app/plugins/datasource/loki/datasource.ts [hunk 2]
```diff
@@ -453,21 +451,12 @@ export class LokiDatasource
...
-  }
-
   /**
    * Given a time range, returns it as Loki parameters.
    * @returns An object containing the start and end times in nanoseconds since the Unix epoch.
    */
-  getTimeRangeParams(timeRange?: TimeRange) {
-    const range = timeRange ?? this.getTimeRange();
-    return { start: range.from.valueOf() * NS_IN_MS, end: range.to.valueOf() * NS_IN_MS };
+  getTimeRangeParams(timeRange: TimeRange) {
```
Comment (matyax) [inline public/app/plugins/datasource/loki/datasource.ts:458]: <3

File: public/app/plugins/datasource/loki/querybuilder/components/LokiQueryBuilder.tsx [hunk 3]
```diff
@@ -99,13 +99,14 @@ export const LokiQueryBuilder = React.memo<Props>(
     useEffect(() => {
       const onGetSampleData = async () => {
         const lokiQuery = { expr: lokiQueryModeller.renderQuery(query), refId: 'data-samples' };
-        const series = await datasource.getDataSamples(lokiQuery);
-        const sampleData = { series, state: LoadingState.Done, timeRange: getDefaultTimeRange() };
+        const range = timeRange ?? getDefaultTimeRange();
```
Comment (matyax) [inline public/app/plugins/datasource/loki/querybuilder/components/LokiQueryBuilder.tsx:102]: On line 37 you can also:

```jsx
({ datasource, query, onChange, onRunQuery, showExplain, timeRange = getDefaultTimeRange() }) => {
```

File: public/app/plugins/datasource/loki/querybuilder/components/UnwrapParamEditor.tsx [hunk 4]
```diff
@@ -53,15 +54,17 @@ export function UnwrapParamEditor({
 
 async function loadUnwrapOptions(
   query: LokiVisualQuery,
-  datasource: LokiDatasource
+  datasource: LokiDatasource,
+  timeRange?: TimeRange
```
Comment (matyax) [inline public/app/plugins/datasource/loki/querybuilder/components/UnwrapParamEditor.tsx:58]:
```suggestion
  timeRange = getDefaultTimeRange()
```

**Review by matyax — APPROVED** [review]
Very nice! 👏

File: public/app/plugins/datasource/loki/querybuilder/components/UnwrapParamEditor.tsx (hunk 4 above)
Comment (ivanahuckova) [reply public/app/plugins/datasource/loki/querybuilder/components/UnwrapParamEditor.tsx]: Fixed in https://github.com/grafana/grafana/pull/78642/commits/acb281e84d70dfbf84feed52713677e5024f4091

**Pull request merged**